envs/monkey_zoo/blackbox/test_blackbox.py

```diff
@@ -326,6 +326,76 @@ def test_agent__cannot_access_nonagent_endpoints(island):
     )
 
 
+def test_unauthenticated_user_cannot_access_API(island):
+    island_requests = MonkeyIslandRequests(island)
+
+    assert (
+        island_requests.post(AGENT_EVENTS_ENDPOINT, data=None).status_code
+        == HTTPStatus.UNAUTHORIZED
+    )
+    assert (
+        island_requests.post(AGENT_HEARTBEAT_ENDPOINT, data=None).status_code
+        == HTTPStatus.UNAUTHORIZED
+    )
+    assert island_requests.put(PUT_LOG_ENDPOINT, data=None).status_code == HTTPStatus.UNAUTHORIZED
+    assert island_requests.get(GET_AGENT_PLUGINS_ENDPOINT).status_code == HTTPStatus.UNAUTHORIZED
+    assert (
+        island_requests.get("/api/agent-plugins/plugin-type/plugin-name/manifest").status_code
+        == HTTPStatus.UNAUTHORIZED
+    )
+    assert island_requests.get(GET_AGENT_SIGNALS_ENDPOINT).status_code == HTTPStatus.UNAUTHORIZED
+    assert (
+        island_requests.post(GET_AGENTS_ENDPOINT, data=None).status_code == HTTPStatus.UNAUTHORIZED
+    )
+    assert island_requests.get(GET_AGENT_EVENTS_ENDPOINT).status_code == HTTPStatus.UNAUTHORIZED
+    assert island_requests.get(PUT_LOG_ENDPOINT).status_code == HTTPStatus.UNAUTHORIZED
+    assert (
+        island_requests.post(TERMINATE_AGENTS_ENDPOINT, data=None).status_code
+        == HTTPStatus.UNAUTHORIZED
+    )
```

Comment:
Long test. At least let's divide them by POST and GET.

Reply:
I think an all around better approach would be to publish something like an OpenAPI spec and then redesign this test to programatically try all endpoints.

Reply:
Breaking these up could result in a large number of login requests which is unnecessary. A whole lot of refactoring needs to happen in this file anyway. I say we just push it until later.